EVEREST-213 Added docs (#80)
percona/percona-everest-backend: README.md

## Welcome to Percona Everest Backend

> Percona Everest is an open-source Database as a Service solution that helps automate day 1/day 2 operations of Postgres/MySQL/MongoDB databases running in Kubernetes clusters.

Reviewer: ```suggestion
Percona Everest is an open-source Database as a Service solution that automates day-1/day-2 operations for Postgres/MySQL/MongoDB databases running in Kubernetes clusters.
```

## Getting started 

Percona Everest has two main components that help you set up the environment:

1. [CLI](https://github.com/percona/percona-everest-cli) implements the installation of required components for Everest to work
2. Backend that implements DBaaS features

You can start playing with Everest using the following way

```sh
wget https://raw.githubusercontent.com/percona/percona-everest-backend/main/quickstart.yml
docker-compose -f quickstart.yml up -d
```
It will spin up the backend/frontend that will be accessible on the http://127.0.0.1:8080 address

> ## Creating Kubernetes cluster

Reviewer: should it be a prerequisite?

> You can try creating an EKS cluster using [DBaaS Infrastructure creator](https://percona.community/labs/dbaas-creator/). Also, it works on minikube/kind/k3d

Reviewer: 1. We should probably expand a bit here, that everest does not handle k8s provisioning.
2. We should not point to EKS right away, but either state that any Kubernetes flavor would do, but here are the requirements (like storage is a must)
3. I would not point to dbaas-creator and again be more generalistic here. Or have it as the last item in the list.

> ## Installing everything needed into the Kubernetes cluster

Reviewer: shall we move it to getting started section? Otherwise the flow is a bit weird.

Reviewer 2: I agree. It would be a good idea to move it getting started section.

To install all required operators in the headless mode you can run these commands

```
git clone git@github.com:percona/percona-everest-cli
cd percona-everest-cli
go run cmd/everest/main.go install operators --backup.enable=false --everest.endpoint=http://127.0.0.1:8080 --monitoring.enable=false --name=minikube --operator.mongodb=true --operator.postgresql=true --operator.xtradb-cluster=true --skip-wizard
```

> However, you can try running it using the wizard

Reviewer: ```suggestion
You can also run it using the wizard:
```

## Welcome to Percona Everest Backend

Percona Everest is an open source Database-as-a-Service solution that automates day-one and day-two operations for Postgres, MySQL, and MongoDB databases within Kubernetes clusters.

## Prerequisites

> A Kubernetes cluster is available for public use, but we do not offer support for creating one.```

Reviewer 3: ```suggestion
A Kubernetes cluster is available for public use, but we do not offer support for creating one.

## Creating Kubernetes cluster

You must have a publicly accessible Kubernetes cluster to use Everest. EKS or GKE is recommended, as it may be difficult to make it work with local installations of Kubernetes such as minikube, kind, k3d, or similar products. Everest does not help with spinning up a Kubernetes cluster but assists with installing all the necessary components for Everest to run.

## Getting started 

The Percona Everest has two primary components that assist you in creating the environment:

1. [CLI](https://github.com/percona/percona-everest-cli), which installs Everest's required components. 
2. Backend, which installs DBaaS features.

To start using Everest, use the following commands:

```sh
wget https://raw.githubusercontent.com/percona/percona-everest-backend/main/quickstart.yml
docker-compose -f quickstart.yml up -d
```
This will spin up the backend/frontend, accessible at http://127.0.0.1:8080.

> **Everest provisioning**

Reviewer 3: ```suggestion
### Everest provisioning
```

Download the latest release of [everestctl](https://github.com/percona/percona-everest-cli/releases) command for your operating system and run the following command to install all required operators in headless mode:

```
./everestctl install operators --backup.enable=false --everest.endpoint=http://127.0.0.1:8080 --monitoring.enable=false --name=minikube --operator.mongodb=true --operator.postgresql=true --operator.xtradb-cluster=true --skip-wizard
```
Alternatively, use the wizard to run it:

```
✗ ./everestctl install operators
? Everest URL http://127.0.0.1:8080
? Choose your Kubernetes Cluster name k3d-everest-dev
? Do you want to enable monitoring? No
? Do you want to enable backups? No
? What operators do you want to install? MySQL, MongoDB, PostgreSQL
```

> Once provisioning is complete, you can visit http://127.0.0.1:8080 to create your first database cluster!```

Reviewer 3: ```suggestion
Once provisioning is complete, you can visit http://127.0.0.1:8080 to create your first database cluster!
```

## Known limitations

> - Currently, Everest only allows for the basic creation of database clusters without monitoring integration or backup/restore support. However, we will be adding this functionality in the near future.```

Reviewer 3: ```suggestion
- Currently, Everest only allows for the basic creation of database clusters without monitoring integration or backup/restore support. However, we will be adding this functionality in the near future.

## Welcome to Percona Everest Backend

Percona Everest is an open-source Database as a Service solution that automates day-1/day-2 operations for Postgres/MySQL/MongoDB databases running in Kubernetes clusters.

## Prerequisites

Publicly accessible Kubernetes cluster ready to use. We do not provide guidance or solutions for spinning up Kubernetes clusters.

## Creating Kubernetes cluster

> You need to have a publicly accessible Kubernetes cluster to try Everest. You can go with EKS, GKE. You might find it challenging to make it work using local installations of Kubernetes like minikube, kind, k3d or similar products.  Everest does not help you to spin-up Kubernetes cluster but it helps you to install all required components for Everest to run.

Reviewer 3: ```suggestion
You must have a publicly accessible Kubernetes cluster to use Everest. EKS or GKE is recommended, as it may be difficult to make it work with local installations of Kubernetes such as minikube, kind, k3d, or similar products. Everest does not help with spinning up a Kubernetes cluster but assists with installing all the necessary components for Everest to run.
```

## Getting started 

> Percona Everest has two main components that help you set up the environment:

Reviewer 3: ```suggestion
The Percona Everest has two primary components that assist you in creating the environment:
```

> 1. [CLI](https://github.com/percona/percona-everest-cli) implements the installation of required components for Everest to work

Reviewer 3: ```suggestion
1. [CLI](https://github.com/percona/percona-everest-cli), which installs Everest's required components. 
```

> 2. Backend that implements DBaaS features

Reviewer 3: ```suggestion
2. Backend, which installs DBaaS features.
```

> You can start playing with Everest using the following way

Reviewer 3: ```suggestion
To start using Everest, use the following commands:
```

```sh
wget https://raw.githubusercontent.com/percona/percona-everest-backend/main/quickstart.yml
docker-compose -f quickstart.yml up -d
```

> It will spin up the backend/frontend that will be accessible on the http://127.0.0.1:8080 address

Reviewer 3: ```suggestion
This will spin up the backend/frontend, accessible at http://127.0.0.1:8080.
```

**Everest provisioning**

> To install all required operators in the headless mode you can run these commands

Reviewer 3: ```suggestion
Run the following commands to install all required operators in headless mode:
```

```
git clone git@github.com:percona/percona-everest-cli
cd percona-everest-cli
go run cmd/everest/main.go install operators --backup.enable=false --everest.endpoint=http://127.0.0.1:8080 --monitoring.enable=false --name=minikube --operator.mongodb=true --operator.postgresql=true --operator.xtradb-cluster=true --skip-wizard
```

> You can also run it using the wizard:

Reviewer 3: ```suggestion
Alternatively,  use wizard to run it:
```

Reviewer 3: does "it" refer to Everest here? If so we should e specific as it's a bit confusing right now

```
✗ go run cmd/everest/main.go install operators
? Everest URL http://127.0.0.1:8080
? Choose your Kubernetes Cluster name k3d-everest-dev
? Do you want to enable monitoring? No
? Do you want to enable backups? No
? What operators do you want to install? MySQL, MongoDB, PostgreSQL
```

> Once provisioning will be finished you can go to http://127.0.0.1:8080 and create your first database cluster!

Reviewer 3: ```suggestion
Once provisioning is complete, you can visit http://127.0.0.1:8080 to create your first database cluster!```

## Known limitations

> 1. It supports only the basic creation of database clusters without monitoring integration and backup/restore support. We will add this support soon

Reviewer 3: ```suggestion
- Currently, Everest only allows for the basic creation of database clusters without monitoring integration or backup/restore support. However, we will be adding this functionality in the near future.```

Reviewer 3: Numbered lists are for chronological items.

> 2. It supports only one Kubernetes cluster on the user interface, however registering multiple Kubernetes clusters is possible.

Reviewer 3: ```suggestion
- It is possible to register multiple Kubernetes clusters, but the user interface only supports one.
```

> 3. No authentication/Access control features. You can configure it with the existing authentication solution you're using 

Reviewer 3: ```suggestion
- There are no authentication or access control features, but you can integrate Everest with your existing solution.
```
